src/chrome/content/rules/Stack-Exchange.xml

```diff
@@ -1,177 +1,128 @@
+	<target host="www.stackoverflow.com" />
+		
+	<!-- Superuser -->
 	<target host="superuser.com" />
-	<target host="*.superuser.com" />
+	<target host="meta.superuser.com" />
+	<target host="www.superuser.com" />
+	
+	<!-- Misc -->
+	<target host="sstatic.net" />
+	<target host="cdn.sstatic.net" />
+	<target host="stackapps.com" />
 	<target host="teststackoverflow.com" />
-	<target host="*.teststackoverflow.com" />
-
+	<target host="www.teststackoverflow.com" />
 
-	<!--	Not secured by server:
-					-->
-	<!--securecookie host="^\.serverfault\.com$" name="^prov$" /-->
-	<!--securecookie host="^\.stackexchange\.com$" name="^prov$" /-->
-	<!--securecookie host="^stackoverflow\.com$" name="^rk-\d+$" /-->
-	<!--securecookie host="^\.stackoverflow\.com$" name="^prov$" /-->
+<!-- Securecookies -->
 
-	<securecookie host="^\.sstatic\.net$" name=".+" />
-	<securecookie host="\.stackexchange\.com$" name=".+" />
-	<securecookie host="^stackoverflow\.com$" name=".+" />
-	<securecookie host="^\.teststackoverflow\.com$" name=".+" />
+	<!-- Blogoverflow -->
+	<securecookie host="^blogoverflow\.com$" name=".+" />
 
+	<!-- Askubuntu -->
+	<securecookie host="^askubuntu\.com$" name=".+" />
+	<securecookie host="^meta\.askubuntu.com$" name=".+" />
 
-	<rule from="^http://(?:www\.)?blogoverflow\.com/$"
-		to="https://stackexchange.com/blogs" />
+	<!-- Mathoverflow -->
+	<securecookie host="^mathoverflow\.net$" name=".+" />
+	<securecookie host="^meta\.mathoverflow.net$" name=".+" />
 	
-	<rule from="^http://([^\/]+)\.blogoverflow\.com/"
-		to="https://$1.blogoverflow.com/" />
-
-	<rule from="^http://(?:(or\.)?cdn\.)?sstatic\.net/"
-		to="https://$1cdn.sstatic.net/" />
-
-	<rule from="^http://cdn-c(areers|hat)\.sstatic\.net/"
-		to="https://cdn-c$1.sstatic.net/" />
-
-	<rule from="^http://(?:www\.)?stackapps\.com/"
-		to="https://stackapps.com/" />
-
-	<rule from="^http://((?:\w+|discuss\.area51)\.)?stackexchange\.com/"
-		to="https://$1stackexchange.com/" />
-
-	<rule from="^http://((?:blog|www|meta|careers)\.)?stackoverflow\.com/"
-		to="https://$1stackoverflow.com/" />
-
-	<rule from="^http://app\.stacktack\.com/"
-		to="https://s3.amazonaws.com/stacktackapp/" />
-
-	<rule from="^http://(www\.|meta\.)?superuser\.com/"
-		to="https://$1superuser.com/" />
-
-	<rule from="^http://(\w\w)\.stackoverflow\.com/"
-		to="https://$1.stackoverflow.com/" />
-
-	<rule from="^http://meta\.(\w\w)\.stackoverflow\.com/"
-		to="https://meta.$1.stackoverflow.com/" />
+	<!-- Serverfault -->
+	<securecookie host="^clc\.serverfault\.com$" name=".+" />
+	<securecookie host="^meta\.serverfault\.com$" name=".+" />
+	<securecookie host="^serverfault\.com$" name=".+" />
+	
+	<!-- Stackauth -->
+	<securecookie host="^stackauth\.com$" name=".+" />
+	<securecookie host="^dev\.stackauth\.com$" name=".+" />
+	
+	<!-- Stackexchange -->
+	<securecookie host="^stackexchange\.com$" name=".+" />
+	<securecookie host="^([\w-]+)\.stackexchange\.com$" name=".+" />
```

Comment:
Yes, there are cookies that will get secured. I didn't check all subdomains, but I guess it wouldn't be bad to have them all there even if there were no cookies for some of them.

Again, I don't know a lot about cookies but I noticed something that seems odd to me. In many cases, both secure and unsecure cookies are set at the same time for pages secured by HTTPS-E, is this normal? If I disable HTTPS-E and load the same pages, then only unsecure cookies are created. Some cookies also appear never to be secured, even though they should be according to the ruleset.

Is there any documentation explaining how HTTPS-E interacts with cookies, and what the best practices are for creating and testing the securecookie instructions?